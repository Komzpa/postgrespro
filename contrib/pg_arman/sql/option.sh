#!/bin/bash

#============================================================================
# This is a test script for options of pg_arman.
#============================================================================

# Load common rules
. sql/common.sh option

cleanup

echo '###### COMMAND OPTION TEST-0001 ######'
echo '###### help option ######'
pg_arman --help;echo $?
echo ''

echo '###### COMMAND OPTION TEST-0002 ######'
echo '###### version option ######'
pg_arman --version;echo $?
echo ''

echo '###### COMMAND OPTION TEST-0003 ######'
echo '###### backup command failure without backup path option ######'
pg_arman backup -A ${ARCLOG_PATH} -b full -p ${TEST_PGPORT};echo $?
echo ''

echo '###### COMMAND OPTION TEST-0004 ######'
echo '###### backup command failure without backup mode option ######'
pg_arman backup -B ${BACKUP_PATH} -A ${ARCLOG_PATH} -p ${TEST_PGPORT};echo $?
echo ''

echo '###### COMMAND OPTION TEST-0005 ######'
echo '###### backup command failure with invalid backup mode option ######'
pg_arman backup -B ${BACKUP_PATH} -A ${ARCLOG_PATH} -b bad -p ${TEST_PGPORT};echo $?
echo ''

echo '###### COMMAND OPTION TEST-0006 ######'
echo '###### delete failure without archive path ######'
pg_arman delete -B ${BACKUP_PATH};echo $?
echo ''

echo '###### COMMAND OPTION TEST-0007 ######'
echo '###### delete failure without DATE ######'
pg_arman delete -B ${BACKUP_PATH} -A ${ARCLOG_PATH};echo $?
echo ''

init_backup

echo '###### COMMAND OPTION TEST-0008 ######'
echo '###### syntax error in pg_arman.ini ######'
echo " = INFINITE" >> ${BACKUP_PATH}/pg_arman.ini
pg_arman backup -B ${BACKUP_PATH} -A ${ARCLOG_PATH} -p ${TEST_PGPORT};echo $?
echo ''

echo '###### COMMAND OPTION TEST-0009 ######'
echo '###### invalid value in pg_arman.ini ######'
init_catalog
echo "BACKUP_MODE=" >> ${BACKUP_PATH}/pg_arman.ini
pg_arman backup -B ${BACKUP_PATH} -A ${ARCLOG_PATH} -p ${TEST_PGPORT};echo $?
echo ''

echo '###### COMMAND OPTION TEST-0010 ######'
echo '###### invalid value in pg_arman.ini ######'
init_catalog
echo "KEEP_DATA_GENERATIONS=TRUE" >> ${BACKUP_PATH}/pg_arman.ini
pg_arman backup -B ${BACKUP_PATH} -A ${ARCLOG_PATH} -b full -p ${TEST_PGPORT};echo $?
echo ''

echo '###### COMMAND OPTION TEST-0011 ######'
echo '###### invalid value in pg_arman.ini ######'
init_catalog
echo "SMOOTH_CHECKPOINT=FOO" >> ${BACKUP_PATH}/pg_arman.ini
pg_arman backup -B ${BACKUP_PATH} -A ${ARCLOG_PATH} -b full -p ${TEST_PGPORT};echo $?
echo ''

echo '###### COMMAND OPTION TEST-0012 ######'
echo '###### invalid option in pg_arman.ini ######'
init_catalog
echo "TIMELINEID=1" >> ${BACKUP_PATH}/pg_arman.ini
pg_arman backup -B ${BACKUP_PATH} -A ${ARCLOG_PATH} -b full -p ${TEST_PGPORT};echo $?
echo ''

echo '###### COMMAND OPTION TEST-0013 ######'
echo '###### check priority of several pg_arman.ini files ######'
init_catalog
mkdir -p ${BACKUP_PATH}/conf_path_a
echo "BACKUP_MODE=ENV_PATH" > ${BACKUP_PATH}/pg_arman.ini
echo "BACKUP_MODE=ENV_PATH_A" > ${BACKUP_PATH}/conf_path_a/pg_arman.ini
pg_arman backup -B ${BACKUP_PATH} -A ${ARCLOG_PATH} -p ${TEST_PGPORT};echo $?
echo ''

# clean up the temporal test data
pg_ctl stop -m immediate > /dev/null 2>&1
rm -fr ${PGDATA_PATH}
rm -fr ${BACKUP_PATH}
rm -fr ${ARCLOG_PATH}
rm -fr ${TBLSPC_PATH}
