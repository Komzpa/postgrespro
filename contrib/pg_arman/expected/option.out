\! bash sql/option.sh
###### COMMAND OPTION TEST-0001 ######
###### help option ######
pg_arman manage backup/recovery of PostgreSQL database.

Usage:
  pg_arman OPTION init
  pg_arman OPTION backup
  pg_arman OPTION restore
  pg_arman OPTION show [DATE]
  pg_arman OPTION validate [DATE]
  pg_arman OPTION delete DATE

Common Options:
  -D, --pgdata=PATH         location of the database storage area
  -A, --arclog-path=PATH    location of archive WAL storage area
  -B, --backup-path=PATH    location of the backup storage area
  -c, --check               show what would have been done
  -j, --threads=NUM         num threads for backup and restore
  --stream                  use stream for save/restore WAL during backup
  --progress                show progress copy files

Backup options:
  -b, --backup-mode=MODE    full,page,ptrack
  -C, --smooth-checkpoint   do smooth checkpoint before backup
  --validate                validate backup after taking it
  --keep-data-generations=N keep GENERATION of full data backup
  --keep-data-days=DAY      keep enough data backup to recover to DAY days age
  --disable-ptrack-clear    disable clear ptrack for postgres without ptrack
  --backup-pg-log           start backup pg_log directory

Restore options:
  --recovery-target-time    time stamp up to which recovery will proceed
  --recovery-target-xid     transaction ID up to which recovery will proceed
  --recovery-target-inclusive whether we stop just after the recovery target
  --recovery-target-timeline  recovering into a particular timeline

Catalog options:
  -a, --show-all            show deleted backup too

Connection options:
  -d, --dbname=DBNAME       database to connect
  -h, --host=HOSTNAME       database server host or socket directory
  -p, --port=PORT           database server port
  -U, --username=USERNAME   user name to connect as
  -w, --no-password         never prompt for password
  -W, --password            force password prompt

Generic options:
  -q, --quiet               don't write any messages
  -v, --verbose             verbose mode
  --help                    show this help, then exit
  --version                 output version information, then exit

Read the website for details. <https://github.com/michaelpq/pg_arman>
Report bugs to <https://github.com/michaelpq/pg_arman/issues>.
1

###### COMMAND OPTION TEST-0002 ######
###### version option ######
pg_arman 0.1
1

###### COMMAND OPTION TEST-0003 ######
###### backup command failure without backup path option ######
ERROR: required parameter not specified: BACKUP_PATH (-B, --backup-path)
1

###### COMMAND OPTION TEST-0004 ######
###### backup command failure without backup mode option ######
ERROR: Required parameter not specified: BACKUP_MODE (-b, --backup-mode)
1

###### COMMAND OPTION TEST-0005 ######
###### backup command failure with invalid backup mode option ######
ERROR: invalid backup-mode "bad"
1

###### COMMAND OPTION TEST-0006 ######
###### delete failure without archive path ######
ERROR: delete command needs ARCLOG_PATH (-A, --arclog-path) to be set
1

###### COMMAND OPTION TEST-0007 ######
###### delete failure without DATE ######
ERROR: required delete range option not specified: delete DATE
1

###### COMMAND OPTION TEST-0008 ######
###### syntax error in pg_arman.ini ######
WARNING: syntax error in " = INFINITE"
ERROR: Required parameter not specified: BACKUP_MODE (-b, --backup-mode)
1

###### COMMAND OPTION TEST-0009 ######
###### invalid value in pg_arman.ini ######
ERROR: invalid backup-mode ""
1

###### COMMAND OPTION TEST-0010 ######
###### invalid value in pg_arman.ini ######
ERROR: option --keep-data-generations should be a 32bit signed integer: 'TRUE'
1

###### COMMAND OPTION TEST-0011 ######
###### invalid value in pg_arman.ini ######
ERROR: option -C, --smooth-checkpoint should be a boolean: 'FOO'
1

###### COMMAND OPTION TEST-0012 ######
###### invalid option in pg_arman.ini ######
ERROR: invalid option "TIMELINEID"
1

###### COMMAND OPTION TEST-0013 ######
###### check priority of several pg_arman.ini files ######
ERROR: invalid backup-mode "ENV_PATH"
1

